src/vstest.console/Processors/ResultsDirectoryArgumentProcessor.cs

```diff
@@ -139,20 +139,14 @@ public void Initialize(string argument)
 
             try
             {
-                // to check valid directory path
-                if (!string.IsNullOrEmpty(argument) && !IsValidPath(argument) )
-                {
-                    throw new ArgumentException("Illegal characters in path.");
-                }
-
-                var di = new DirectoryInfo(argument);
-
                 if (!Path.IsPathRooted(argument))
```

Comment:
`Path.IsPathRooted` will tell you if the path is fixed to a drive, not if it is relative or not. `D:foo.txt` is relative to wherever the working directory is. `Path.IsPathFullyQualified()` was added in core to tell you if the location would change based on the current working directory. Fully qualified, however, doesn't mean normalized, so if you're doing any manipulation on the path you should always call `Path.GetFullPath()`.

In this case it is pointless to call `Path.GetFullPath()` as `Directory.CreateDirectory()` will do that anyway.